File: insertionSort.cpp
void insertionSort(int arr[], int length){
    int i, j, curValue;
    for (i = 1; i < length; i++){
        j = i;
        curValue = arr[j];
        while (j > 0 && curValue < arr[j-1]){
            arr[j] = arr[j-1];
            j--;
        }
        arr[j] = curValue;
    }
}

File: insertionSort.java
class App {
    void insertionSort(int[] arr){
        int i, j, temp;
        for (i = 1; i < arr.length; i++){
            j = i;
            while (j > 0 && arr[j-1] > arr[j]){
                temp = arr[j-1];
                arr[j-1] = arr[j];
                arr[j] = temp;
                j--;
            }
        }
    }
}
